rasa_core/channels/slack.py

```diff
@@ -129,13 +133,19 @@ def _is_user_message(slack_event):
 
     @staticmethod
     def _is_button_reply(slack_event):
-        return (slack_event.get('payload') and
-                slack_event['payload'][0] and
-                'name' in slack_event['payload'][0])
+        pay = json.loads(slack_event['payload']);
+        evaluation = (pay['type'] == u"interactive_message") and (pay['actions'][0]['type'] == u"button");
```

Comment:
could you please change this into a couple of functions like `is_interactive` and `is_button` ?

Reply:
Feck ... can't attach anything (.gz or .py) 

reformatted into functions for you

    @staticmethod
    def is_interactive_message(payload):
        return (payload['type'] == u"interactive_message")

    @staticmethod
    def is_button(payload):
        return (payload['actions'][0]['type'] == u"button")

    @staticmethod
    def _is_button_reply(slack_event):
        payload = json.loads(slack_event['payload']);
        return is_interactive_message(payload) and is_button(payload)

Reply:
@EPedrotti  hope this helps, I cant attach files to this for some reason (nothing seems to get uploaded)  so I hope you can make sense of the code snippets